cmd/cluster/etcd_health.go

```diff
@@ -0,0 +1,239 @@
+
+	err := ControlplaneNodeStatus(kubeCli)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	podlist, err := EtcdPodStatus(kubeCli, flags)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	err = EtcdCrStatus(kubeCli)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	var kubeconfig *string
+
+	if home := homedir.HomeDir(); home != "" {
+		kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "(optional) absolute path to the kubeconfig file")
+	} else {
+		kubeconfig = flag.String("kubeconfig", "", "absolute path to the kubeconfig file")
+	}
+	flag.Parse()
+
+	// use the current context in kubeconfig
+	kconfig, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	// create the clientset
+	clientset, err := kubernetes.NewForConfig(kconfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	MatchPodName := false
+	var etcdPodName string
+	for !MatchPodName {
+		fmt.Print("Enter Etcd Pod Name: ")
+		fmt.Scan(&etcdPodName)
+		for _, pods := range podlist.Items {
+			if etcdPodName == pods.Name {
+				MatchPodName = true
+			}
+		}
+	}
+
+	for _, v := range etcdctlCmd {
+		output, err := Etcdctlhealth(kconfig, clientset, v, etcdPodName)
+		if err != nil {
+			fmt.Println(err)
+		}
+		fmt.Printf("$ %s\n", v)
+		fmt.Println(output)
+	}
+	return nil
+}
+
+func ControlplaneNodeStatus(kubeCli client.Client) error {
+	nodeList := &corev1.NodeList{}
+	if err := kubeCli.List(context.TODO(), nodeList, client.MatchingLabels{MasterNodeLabel: ""}); err != nil {
+		return err
+	}
+
+	fmt.Println("+----------------------------------------------------------------+")
+	fmt.Printf("|                CONTROLPLANE NODE STATUS                        |\n")
+	fmt.Println("+----------------------------------------------------------------+")
+
+	for _, node := range nodeList.Items {
+		for _, condition := range node.Status.Conditions {
+			if condition.Type == corev1.NodeReady {
+				fmt.Printf("%s\t%s\t%s\n", node.Name, corev1.NodeReady, condition.Status)
+			}
+		}
+	}
+	return nil
+}
+
+func EtcdPodStatus(kubeCli client.Client, flags *genericclioptions.ConfigFlags) (*corev1.PodList, error) {
+
+	pods := &corev1.PodList{}
+	if err := kubeCli.List(context.TODO(), pods, client.InNamespace(EtcdNamespceNAME), client.MatchingLabels{EtcdPodMatchLabelName: EtcdPodMatchValueName}); err != nil {
+		return nil, err
+	}
+
+	fmt.Println("+----------------------------------------------------------------+")
+	fmt.Printf("|               ETCD POD STATUS                                  |\n")
+	fmt.Println("+----------------------------------------------------------------+")
+
+	for _, pods := range pods.Items {
+		containerReadyCount := 0
+		for _, containers := range pods.Status.ContainerStatuses {
+			if containers.Ready {
+				containerReadyCount++
+			}
+		}
+		fmt.Printf("%s\t%v/%v\n", pods.Name, containerReadyCount, len(pods.Status.ContainerStatuses))
```

Comment:
A similar thought here to the one I had concerning the node status. Rather than print the status of every container, just list the ones where the status is abnormal.